### ISSUE TITLE
[CLI] Tensorboard integration bug with resetting x-axis through global_step

### ISSUE THREAD

**Issue by vwxyzjn** [wandb/client#1997]
**Description**

When using the tensorboard integration, the latest version `wandb==0.10.23` does not seem to handle `global_step` properly when it was reset to 0, whereas the older version of `wandb==0.9.5` seems to have recorded correctly. 

Consider a simple code sample:

```python
import numpy as np
import random
import time
from torch.utils.tensorboard import SummaryWriter
import wandb

if __name__ == "__main__":
    # setup the environment
    experiment_name = f"wandb=={wandb.__version__}__{int(time.time())}"
    run = wandb.init(project="cleanRL", entity=None, 
                     sync_tensorboard=True, name=experiment_name, 
                     save_code=True,)
    writer = SummaryWriter(f"runs/{experiment_name}")
    for global_step in range(100):
        writer.add_scalar("loss1", np.random.rand(1)[0], global_step)
    for global_step in range(100):
        writer.add_scalar("loss2", np.random.rand(1)[0], global_step)
```

In particular, notice that the loss2 is recorded when the `global_step` has been reset. When visualizing, it shows like this: 

![image](https://user-images.githubusercontent.com/5555347/112222712-4fff8800-8bff-11eb-9965-80d61c91bd04.png)

If we visualize the loss using `global_step` as the x-axis, the runs recorded with `wandb==0.10.23` seem to break while `wandb==0.9.5` seems to work correctly. Also the [tensorboard page](https://wandb.ai/costa-huang/cleanRL/runs/3b4feosq/tensorboard?workspace=user-costa-huang) for `wandb==0.10.23` also displays correctly.

![image](https://user-images.githubusercontent.com/5555347/112223685-a1f4dd80-8c00-11eb-8fc6-00f5973e9380.png)

See report for details. https://wandb.ai/costa-huang/cleanRL/reports/Tensorboard-global_step-misalignment--Vmlldzo1NTQzMzA

`global_step` in tensorboard is a way to customize the x-axis and quite helpful. Without it, it would look like the following two cases if I did anything with tensorboard:

![image](https://user-images.githubusercontent.com/5555347/112223237-05cad680-8c00-11eb-9a9e-b26b561e793c.png)
![image](https://user-images.githubusercontent.com/5555347/112223358-30b52a80-8c00-11eb-9018-a09fb82e7e3c.png)

Neither is desirable (the first image has a bug while the second does not align the x-axis).

Hope this is helpful :)

**Comment by ariG23498** [comment]
Hey @vwxyzjn 
Wandb Version: **0.10.23**
I tried reproducing the issue with the code snippet that you had sent in.
This is the dashboard for the run: https://wandb.ai/repro/GH-1997
I do not see the `loss2` metric breaking in any way. In fact both the metrics start from the same point in the x-axis when I am plotting them against `global_step`.

I made a small change to the code, but that should account to anything big.
```python
experiment_name = f"wandb=={wandb.__version__}__{int(time.time())}"
wandb.tensorboard.patch(root_logdir=f'runs/{experiment_name}')
run = wandb.init(entity="repro", project="GH-1997")

writer = SummaryWriter(f"runs/{experiment_name}")
for global_step in range(100):
    writer.add_scalar("loss1", np.random.rand(1)[0], global_step)
for global_step in range(100):
    writer.add_scalar("loss2", np.random.rand(1)[0], global_step)
```

Let me know if I have made a mistake or missed something that needs to be looked into.

**Comment by vwxyzjn** [comment]
Hey @ariG23498 Thank you for looking into this, I can confirm this works! It appears having `wandb.tensorboard.patch(root_logdir=f'runs/{experiment_name}')` is crucial.
